bastionzero/service/targets/database.go

```diff
@@ -17,16 +18,24 @@ const (
 
 // CreateDatabaseTargetRequest is used to create a new Database target
 type CreateDatabaseTargetRequest struct {
-	TargetName      string `json:"targetName"`
-	ProxyTargetID   string `json:"proxyTargetId"`
-	RemoteHost      string `json:"remoteHost"`
-	RemotePort      Port   `json:"remotePort"`
-	LocalPort       *Port  `json:"localPort,omitempty"`
-	LocalHost       string `json:"localHost,omitempty"`
-	IsSplitCert     bool   `json:"splitCert,omitempty"`
-	DatabaseType    string `json:"databaseType,omitempty"`
-	EnvironmentID   string `json:"environmentId,omitempty"`
-	EnvironmentName string `json:"environmentName,omitempty"`
+	TargetName    string `json:"targetName"`
+	ProxyTargetID string `json:"proxyTargetId"`
+	RemoteHost    string `json:"remoteHost"`
+	// RemotePort is required for all databases except GCP-hosted ones. For GCP-hosted databases,
+	// Port.Value can be specified but will be ignored when connecting to the database.
+	// If not provided when creating a CGP database target, Port.Value will be set to 0.
+	RemotePort *Port  `json:"remotePort,omitempty"`
```

Comment:
Did you read the comment? Maybe it's not clear, but I made a change in the API to allow it to be null for one specific db type. So, I thought it was appropriate to make this change. We could make it required in this SDK, though, if that makes sense. I'm not sure it does, though.

Reply:
Oh right. Sorry yeah I missed that. It's unfortunate though in Go that this would still be considered a breaking change cause now the field type has changed to a pointer type. Maybe this is evidence that we should be using pointer types everywhere just in case one day a field does become optional by the remote API. Because otherwise, we'll run into this again and cause it to break for those who expected it not to be a pointer type

Reply:
Honestly, you could still keep it as before (no pointer type) and then in the case of GCP where it isn't required, people just won't set the field and that will end up getting serialized to:

```
{"remotePort":{}}
```

but backend should interpret that and set it to `0` the default type of the integer I think?

Reply:
So some extra bytes are sent but at least it is still backwards compatible. And then in future breaking release, we can make the change you did earlier, making it a pointer type.

And then, I think we honestly should have a discussion of just making everything pointer type to prevent this from being a breaking change in the future. I'm starting to realize this is prob why the Go GitHub SDK has everything as pointer types--to prevent a situation like this where remote API makes a field optional at some point in the future.

The only annoying this is it makes using the API a little bit harder to use cause you have to make everything pointers. Luckily, though we do have a helper method to make pointers

https://github.com/bastionzero/bastionzero-sdk-go/blob/02301e76852f04bd3d15aa8f2ad5ab2905c04c4a/bastionzero/bastionzero.go#L301-L304

Reply:
> but backend should interpret that and set it to 0 the default type of the integer I think?
Not quite. The value for `Port.Value` is required by the back end, so it would error out. I'll add a TODO as we discussed offline and will update the comment to something appropriate.